### PR TITLE
Upgrade and refactor vault deployment

### PR DESCRIPTION
**Pull request by larsks**
This pull request replaces our previous vault deployment with one managed using the bank-vaults [vault operator](https://github.com/bank-vaults/vault-operator). The vault operator handles automatically unsealing the vault when necessary (nerc-project/operations#446). At the time we first installed vault, the operator had been abandoned (https://github.com/banzaicloud/bank-vaults), but development has since resumed under [a new organization](https://github.com/bank-vaults).

This deployment uses the current vault release, 1.15.5 (nerc-project/operations#439).

I have organized the vault deployment into three separate "applications":

- `vault/operator` is just the vault operator. It does not create any vault instances.
- `vault/instance` is the actual vault instance.
- `vault/backup-job` is the backup job.

**NB**: 10456 lines of this change are the vault CRD. I should note that we're installing the operator manifests directly because there is no maintained version of this operator in OperatorHub.

### CODE REVIEW

File: vault/backup-job/base/trigger/clusterrolebinding.yaml [hunk 1]
```diff
@@ -9,4 +9,4 @@ roleRef:
 subjects:
 - kind: ServiceAccount
   name: backup-job
-  namespace: vault
+  namespace: default
```
Comment (larsks) [inline vault/backup-job/base/trigger/clusterrolebinding.yaml:12]: This is intentional; when using the namespace `default`, kustomize will update the value when there is a `namespace:` set in `kustomization.yaml`.

**Review by schwesig — APPROVED** [review]
wow, a big clean up. Thanks Lars.

File: vault/operator/base/src/.gitignore
Comment (dystewart) [inline vault/operator/base/src/.gitignore:1]: Is this file included intentionally?

Comment (larsks) [reply vault/operator/base/src/.gitignore:1]: Yes. It prevents accidentally adding the `charts` directory that gets created when you run the Makefile in the parent directory.

**Comment by dystewart** [comment]
LGTM

**Review by dystewart — APPROVED** [review]

**Comment by dystewart** [comment]
Good to see the bank vaults operator is being maintained again

**Pull request merged**